subsys/net/lib/nrf_cloud/src/nrf_cloud_pgps.c

```diff
@@ -74,20 +76,29 @@ struct pgps_index {
 	uint32_t storage_extent;
 	int store_block;
 
-	/* array of pointers to predictions, in sorted time order */
+	/* Array of memory offsets to predictions, in sorted time order.
+	 * If flash device is external, this must be passed
+	 * to read_prediction() to read a copy to a local buffer.
+	 * If flash device is internal, it can be converted directly to
+	 * a pointer.
+	 */
 	struct nrf_cloud_pgps_prediction *predictions[NUM_PREDICTIONS];
 };
 
 static struct pgps_index index;
 
-static pgps_event_handler_t evt_handler;
-/* array of potentially out-of-time-order predictions */
-static uint8_t *storage;
-static uint32_t storage_size;
 static struct stream_flash_ctx stream;
-static uint8_t *write_buf;
+static const struct flash_area *flash;
```

Comment:
A more specific name for this would be nice, maybe `flash_prediction_storage_area` -- and you could rename `flash_dev` to `flash_prediction_storage_dev`